tsdb/store_test.go

```diff
@@ -144,6 +145,64 @@ func TestStore_CreateShard(t *testing.T) {
 	}
 }
 
+// Ensure the store can create a new shard.
+func TestStore_StartupShardProgress(t *testing.T) {
+	t.Parallel()
+
+	test := func(index string) {
+		fmt.Println(index)
+		s := MustOpenStore(index)
+		defer s.Close()
+
+		// Create a new shard and verify that it exists.
+		if err := s.CreateShard("db0", "rp0", 1, true); err != nil {
+			t.Fatal(err)
+		} else if sh := s.Shard(1); sh == nil {
+			t.Fatalf("expected shard")
+		}
+
+		// Create another shard and verify that it exists.
+		if err := s.CreateShard("db0", "rp0", 2, true); err != nil {
+			t.Fatal(err)
+		} else if sh := s.Shard(2); sh == nil {
+			t.Fatalf("expected shard")
+		}
+
+		msl := &mockStartupLogger{}
+
+		// Reopen shard and recheck.
+		if err := s.ReopenWithStartupMetrics(msl); err != nil {
+			t.Fatal(err)
+		} else if sh := s.Shard(1); sh == nil {
+			t.Fatalf("expected shard(1)")
+		} else if sh = s.Shard(2); sh == nil {
+			t.Fatalf("expected shard(2)")
+		}
+
+		if msl.getShardsAdded() != 2 {
+			t.Fatalf("expected 2 shards added, got %d", msl.getShardsAdded())
+		}
+
+		if msl.getShardsCompleted() != 2 {
+			t.Fatalf("expected 2 shards completed, got %d", msl.getShardsCompleted())
+		}
```

Comment:
I'm not sure checking the count adds value when we check not only the exact count but the exact order of the calls down below. If there was a failure, the check for the order of calls down below would be far more helpful in debugging the issue than a simple count.